tests/e2e-net-test.sh

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+
+#
+# This test is executed by github workflows inside the action runners
+#
+
+TRACEE_STARTUP_TIMEOUT=5
+SCRIPT_TMP_DIR=/tmp
+TRACEE_TMP_DIR=/tmp/tracee
+
+info() {
+    echo -n "INFO: "
+    echo $@
+}
+
+error_exit() {
+    echo -n "ERROR: "
+    echo $@
+    exit 1
+}
+
+if [[ $UID -ne 0 ]]
+then
+    error_exit "need root privileges"
+fi
+
+if [[ ! -d ./signatures ]]
+then
+    error_exit "need to be in tracee root directory"
+fi
+
+# run CO-RE IPv4 test only by default
+TESTS=${NETTESTS:=IPv4}
+ISNONCORE=${ISNONCORE:=0}
+DONTSLEEP=${DONTSLEEP:=1}
+
+# randomize start point (for parallel runners)
+if [[ $DONTSLEEP -ne 1 ]]
+then
+  rand=$(( $RANDOM % 10 ))
+  info "sleeping for $rand seconds"
+  sleep $rand
+fi
+
+# startup needs
+rm -rf $TRACEE_TMP_DIR/* || error_exit "could not delete $TRACEE_TMP_DIR"
+git config --global --add safe.directory "*"
+
+info
+info "= ENVIRONMENT ================================================="
+info
+info "KERNEL: $(uname -r)"
+info "NON CO-RE: $ISNONCORE"
+info "CLANG: $(clang --version)"
+info "GO: $(go version)"
+info
+info "= SETUP NETWORK TESTING ENV  =================================="
+info
+timeout --preserve-status 20 ./tests/e2e-net-rules/scripts/setup.sh
+ret=$?
+if [[ $ret -ne 0 ]]; then
+    error_exit "could not setup network namespaces: error $ret"
+fi
+info
+info "= COMPILING TRACEE ============================================"
+info
+# make clean # if you want to be extra cautious
+set -e
+make -j$(nproc) all
+make e2e-net-rules
+set +e
+if [[ ! -x ./dist/tracee-ebpf || ! -x ./dist/tracee-rules ]]
+then
+    error_exit "could not find tracee executables"
+fi
+if [[ $ISNONCORE -eq 1 ]]
+then
+    info "STATE: Compiling non CO-RE eBPF object"
+    make clean-bpf-nocore
+    set -e
+    make install-bpf-nocore
+    set +e
+    export TRACEE_BPF_FILE=$(ls -1tr $TRACEE_TMP_DIR/*tracee.bpf*.o | head -n1)
+fi
+
+# if any test has failed
+anyerror=""
+
+# run tests
+for TEST in $TESTS; do
+
+    info
+    info "= TEST: $TEST =============================================="
+    info
+
+    rm -f $SCRIPT_TMP_DIR/build-$$
+    rm -f $SCRIPT_TMP_DIR/ebpf-$$
+
+    events=$(./dist/tracee-rules --rules-dir ./dist/e2e-net-rules/ --rules $TEST --list-events)
+
+    ./dist/tracee-ebpf \
+        --install-path $TRACEE_TMP_DIR \
+        --cache cache-type=mem \
+        --cache mem-cache-size=512 \
+        --output format:json \
+        --output option:parse-arguments \
+        --output option:detect-syscall \
+        --trace comm=ping,nc \
+        --trace event=$events \
+        2>$SCRIPT_TMP_DIR/ebpf-$$ \
+        | \
+    ./dist/tracee-rules \
+        --rules-dir ./dist/e2e-net-rules/ \
```

Comment:
The lack of the absolute path was raising this error:

```shell
2022/10/20 18:39:35 error opening plugin dist/e2e-net-rules/builtin.so: plugin.Open("dist/e2e-net-rules/builtin.so"): realpath failed`
```

```suggestion
    rulesdir=$(realpath ./dist/e2e-net-rules/)
    events=$(./dist/tracee-rules --rules-dir $rulesdir --rules $TEST --list-events)

    ./dist/tracee-ebpf \
        --install-path $TRACEE_TMP_DIR \
        --cache cache-type=mem \
        --cache mem-cache-size=512 \
        --output format:json \
        --output option:parse-arguments \
        --output option:detect-syscall \
        --trace comm=ping,nc \
        --trace event=$events \
        2>$SCRIPT_TMP_DIR/ebpf-$$ \
        | \
    ./dist/tracee-rules \
        --rules-dir $rulesdir \
```

Reply:
Applying that patch this new one is raising:

```shell
{"level":"fatal","ts":1666302147.5275285,"msg":"app","error":"failed loading CEL signatures: failed loading configs: failed walking dir /home/gg/code/tracee/dist/e2e-net-rules: lstat /home/gg/code/tracee/dist/e2e-net-rules: permission denied"}
```

```shell
ls -l tests 
total 32
drwxr-xr-x 2 gg gg 4096 ago  8 20:46 distro-tester
drwxr-xr-x 3 gg gg 4096 out 20 16:48 e2e-net-rules
-rwxr-xr-x 1 gg gg 4549 out 20 18:42 e2e-net-test.sh
drwxr-xr-x 2 gg gg 4096 out 20 16:48 integration
-rwxr-xr-x 1 gg gg 4615 out 13 20:59 kerneltest.sh
-rwxr-xr-x 1 gg gg  900 set 30 08:36 libbpfgo.sh

ls -l tests/e2e-net-rules 
total 36
-rw-r--r-- 1 gg gg 1789 out 20 16:48 e2e-icmp.go
-rw-r--r-- 1 gg gg 1839 out 20 16:48 e2e-icmpv6.go
-rw-r--r-- 1 gg gg 1863 out 20 16:48 e2e-ipv4.go
-rw-r--r-- 1 gg gg 1905 out 20 16:48 e2e-ipv6.go
-rw-r--r-- 1 gg gg 1860 out 20 16:48 e2e-tcp.go
-rw-r--r-- 1 gg gg 1765 out 20 16:48 e2e-udp.go
-rw-r--r-- 1 gg gg  221 out 20 16:48 export.go
-rw-r--r-- 1 gg gg  330 out 20 16:48 helpers.go
drwxr-xr-x 2 gg gg 4096 out 20 16:48 scripts

```
The folder e2e-net-rules has execution and its files have reading granted. The script enforces its running as root, so it's a odd behaviour. I'll double check it.

Reply:
Running as an unprivileged user it works.

```shell
❯ ./dist/tracee-rules --rules-dir $(realpath ./dist/e2e-net-rules) --rules IPv4 --list-events         
net_packet_ipv4
```

```shell
❯ sudo ./dist/tracee-rules --rules-dir $(realpath ./dist/e2e-net-rules) --rules IPv4 --list-events
{"level":"fatal","ts":1666374093.8744888,"msg":"app","error":"failed loading CEL signatures: failed loading configs: failed walking dir /home/gg/code/tracee/dist/e2e-net-rules: lstat /home/gg/code/tracee/dist/e2e-net-rules: permission denied"}
```

Could it be an ebpf-rules capabilities issue?

Reply:
Solved. Final suggestion:

```suggestion
    rulesdir=$(realpath ./dist/e2e-net-rules/)
    events=$(sudo -u $(logname) ./dist/tracee-rules --rules-dir $rulesdir --rules $TEST --list-events)
    
    ./dist/tracee-ebpf \
        --install-path $TRACEE_TMP_DIR \
        --cache cache-type=mem \
        --cache mem-cache-size=512 \
        --output format:json \
        --output option:parse-arguments \
        --output option:detect-syscall \
        --trace comm=ping,nc \
        --trace event=$events \
        2>$SCRIPT_TMP_DIR/ebpf-$$ \
        | \
    sudo -u $(logname) ./dist/tracee-rules \
        --rules-dir $rulesdir \
```